plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/ScanQueryPageSource.java

```diff
@@ -158,8 +158,9 @@ public Page getNextPage()
             Map<String, Object> document = hit.getSourceAsMap();
 
             for (int i = 0; i < decoders.size(); i++) {
-                String field = columns.get(i).name();
-                decoders.get(i).decode(hit, () -> getField(document, field), columnBuilders[i]);
+                String field = columns.get(i).baseName();
+                List<String> dereferenceNames = columns.get(i).dereferenceNames();
+                decoders.get(i).decode(hit, () -> getField(document, field), columnBuilders[i], dereferenceNames);
```

Comment:
nit: we can extract `columns.get(i)` and use it directly.
```suggestion
                OpenSearchColumnHandle columnHandle = columns.get(i);
                decoders.get(i).decode(hit, () -> getField(document, columnHandle.baseName()), columnBuilders[i], columnHandle.dereferenceNames());
```